File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 3/14/2026, 11:51:14 AM
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        slow = 0
        for fast in range(1,len(nums)):
            if nums[fast] !=nums[slow] and nums[fast] > nums[slow]:
                temp = nums[fast]
                nums[fast] = nums[slow+1]
                nums[slow+1] = temp
                slow +=1
        #print(nums)
        return slow+1

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.java
// Last updated: 3/14/2026, 11:51:18 AM
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length < 2){
            return nums.length;
        }
        int val = 0;
        boolean flag = false;
        for(int i = 0 ; i < nums.length -1;i++){
            for (int j = i+1; j < nums.length; j++) {

                if(nums[i] != nums[j] && nums[i] < nums[j]){
                    int temp = nums[i+1];
                    nums[i+1] = nums[j];
                    nums[j] = temp;
                    val++;
                    if (j == nums.length -1){
                        val++;
                        flag = true;
                    }
                    break;
                }
                if (j == nums.length -1){
                    val++;
                    flag = true;
                }

            }
            if(flag) break;

        }
        return val;
    }
}